_includes/main_nav.html

```diff
@@ -3,23 +3,29 @@
     <a class="navbar-brand" href="/"><img src="/images/bisq-logo.svg" height="30"/></a>
 
     <!-- Language selector for mobile -->
-    {% if page.lang %}
     <div class="dropdown mobile-language-selector more-info-desktop">
       <button class="btn btn-sm dropdown-toggle" type="button" id="dropdownMenuButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
-        {{ page.language }} {{ page.flag }}
+        {% for item in site.data.languages %}
+          {% if  site.lang == item.tag %} {{ item.name }} {{ item.flag }} {% endif %}
+        {% endfor %}
       </button>
+      
       <div class="dropdown-menu" aria-labelledby="dropdownMenuButton">
-        {% assign pages = site.pages | where:"ref", page.ref | sort: 'language' %}
-        {% for page in pages %}
+	   {% assign sorted-items = site.data.languages | sort: "name" %}
+       {% for item in sorted-items %}
           <div class="dropdown-item btn-icon">
             <div class="float-right">
-              <a href="{{ page.url }}" hreflang="{{ page.lang }}">{{ page.language }} {{ page.flag }}</a>
+               {% if item.tag == "en" %}
+               <a href="/">{{ item.name }} {{ item.flag }}</a>
+			   {% else %}
+			   <!-- we use the 'tag' variable in languages.yml to redirect to the correct page -->
```

Comment:
please remove this comment, it's getting output in the source code once per locale